spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -235,9 +235,8 @@ public boolean removeInterceptor(ChannelInterceptor interceptor) {
 	}
 
 	@Override
-	@Nullable
 	public ChannelInterceptor removeInterceptor(int index) {
-		return this.interceptors.remove(index);
+		return Objects.requireNonNull(this.interceptors.remove(index));
```

Comment:
Why is this?
Just fix that `this.interceptors.remove()` to not produce null.
According to the `List.remove(int)` contract it does not return null.
So, that `if (removed != null) {` is wrong